python/src/ops.cpp

```diff
@@ -1318,26 +1322,48 @@ void init_ops(py::module_& m) {
       )pbdoc");
   m.def(
       "scatter",
-      &scatter,
+      [](const array& a,
+         const array& indices,
```

Comment:
Shouldn't this be a `vector<array>` to support multi-axis scattering when axis is a `vector<int>`?

Reply:
As far as I understand it's natively implemented in https://github.com/ml-explore/mlx/blob/main/mlx/ops.cpp#L2183 but I may be wrong.

Reply:
You are right. Ideally, I wanted to have a single binding with a variant that accepts indices as an array or as a vector of arrays. However, the code did not want to compile and I still do not know what is the reason. Anyway, we recently received a comment we should not expose entire ``scatter`` method, but expose ``scatter_add``, ``scatter_prod`` ... Thus, I think it is best to close this PR and open a new one that implements these bindings. I can implement this today, but I might not have enough time to finish tests today. It would be great if you could complete the tests for these bindings.

Reply:
Yes, noted. I suggest we merge this into the branch of the PR anyways to keep track of progress. The general scatter method may turn out to be useful anyways (and it's often present in other libraries as a possible entrypoint).
Let me know what you think and we can merge.